mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```
@@ -0,0 +1,256 @@
+  - !ruby/object:Api::Type::NestedObject
+    name: dedicatedResources
+    description: A description of resources that are dedicated to the DeployedIndex, and that need a higher degree of manual configuration.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'machineSpec'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        immutable: true
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'machineType'
+            description: The type of the machine.
+            immutable: true
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of machine replicas this DeployedModel will be always deployed on. This value must be greater than or equal to 1.
+        required: true
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases. If maxReplicaCount is not set, the default value is minReplicaCount
+        default_from_api: true
+  - !ruby/object:Api::Type::Boolean
+    name: enableAccessLogging
+    description: If true, private endpoint's access logs are sent to Cloud Logging.
+    immutable: true
+    default_value: false
+  - !ruby/object:Api::Type::NestedObject
+    name: deployedIndexAuthConfig
+    description: If set, the authentication is enabled for the private endpoint.
+    immutable: true
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: authProvider
+        description: Defines the authentication provider that the DeployedIndex uses.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: audiences
+            description: The list of JWT audiences. that are allowed to access. A JWT containing any of these audiences will be accepted.
+            immutable: true
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: allowedIssuers
+            immutable: true
+            description: |
+              A list of allowed JWT issuers. Each entry must be a valid Google service account, in the following format: service-account-name@project-id.iam.gserviceaccount.com
+            item_type: Api::Type::String
+  - !ruby/object:Api::Type::Array
+    name: reservedIpRanges
+    immutable: true
+    description: A list of reserved ip ranges under the VPC network that can be used for this DeployedIndex.
+    item_type: Api::Type::String
+  - !ruby/object:Api::Type::String
+    name: deploymentGroup
+    immutable: true
+    default_value: 'default'
+    description: |
+      The deployment group can be no longer than 64 characters (eg: 'test', 'prod'). If not set, we will use the 'default' deployment group. Note: we only support up to 5 deployment groups (not including 'default').
```

It'd be worth including some info about how deploymentGroup interacts with the reservedIpRanges value, as [described in the docs](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.indexEndpoints#DeployedIndex.FIELDS.deployment_group).

```suggestion
      The deployment group can be no longer than 64 characters (eg: 'test', 'prod'). If not set, we will use the 'default' deployment group.
      Creating deployment_groups with reserved_ip_ranges is a recommended practice when the peered network has multiple peering ranges. This creates your deployments from predictable IP spaces for easier traffic administration. Also, one deployment_group (except 'default') can only be used with the same reserved_ip_ranges which means if the deployment_group has been used with reserved_ip_ranges: [a, b, c], using it with [a, b] or [d, e] is disallowed. [See the official documentation here](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.indexEndpoints#DeployedIndex.FIELDS.deployment_group).
      Note: we only support up to 5 deployment groups (not including 'default').
```